staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go

```diff
@@ -258,17 +258,12 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope *RequestScope, forceWatc
 				timeout = time.Duration(float64(minRequestTimeout) * (rand.Float64() + 1.0))
 			}
 			klog.V(3).InfoS("Starting watch", "path", req.URL.Path, "resourceVersion", opts.ResourceVersion, "labels", opts.LabelSelector, "fields", opts.FieldSelector, "timeout", timeout)
-			ctx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
Please move the first commit to a separate PR - I would definitely want more pairs of eye to look at it, we had a number of issues with timeouts in the past...